lib/sinon/assert.js

```diff
@@ -78,6 +78,11 @@ function mirrorPropAsAssertion(name, method, message) {
         var args = slice.call(arguments, 1);
         var failed = false;
 
+        // Hack to be replaced
```

Comment:
1. If this is explicitly to catch matchers, can you use the built-in `sinonMatch.isMatcher(...)` on line 82? 
2. Can you update the comment to say why you're doing this (like some of the posted comment above) instead of this one?